### PR TITLE
Replace Guava by JDK (Partly)

### PR DESCRIPTION
**Pull request by pks-1981**
Replace Guava by JDK (Partly)
https://github.com/wiremock/wiremock/issues/2111

## Submitter checklist

- [ ] Recommended: Join [WireMock Slack](https://slack.wiremock.org/) to get any help in `#help-contributing` or a project-specific channel like `#wiremock-java`
- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [x] Test coverage that demonstrates that the change works as expected
- [ ] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

### CODE REVIEW

**Comment by pks-1981** [comment]
@tomakehurst @oleg-nenashev
MR finished. please look how it will be possible.

**Comment by pks-1981** [comment]
@tomakehurst @oleg-nenashev
Congratulations on the release of 3.0.0 I hope you have time to look

File: src/main/java/com/github/tomakehurst/wiremock/common/ResourceUtil.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class ResourceUtil {
+
+  private ResourceUtil() {}
+
+  public static <T> ClassLoader getLoader(Class<T> className) {
+    return getFirstNonNull(
+        Thread.currentThread().getContextClassLoader(), className.getClassLoader());
```
Comment (oleg-nenashev) [inline src/main/java/com/github/tomakehurst/wiremock/common/ResourceUtil.java:33]: I am not sure it is a right priority order, and most likely not the one expected in `getResource()` where you would start from the class and then, maybe?, try other classloaders

**Review by oleg-nenashev — COMMENTED** [review]
Thanks! We need to discuss the classloader logic itself, but the replacements are good.

@pks-1981 for the WireMock 3 announcements, do you want to be credited as `@pks-1981` or do you prefer a real name or another alias?

File: src/main/java/com/github/tomakehurst/wiremock/common/ResourceUtil.java (hunk 1 above)
Comment (pks-1981) [reply src/main/java/com/github/tomakehurst/wiremock/common/ResourceUtil.java:33]: Hello.
I completely repeated com.google.common.io.Resources

```
public static URL getResource(String resourceName) {
    ClassLoader loader =
        MoreObjects.firstNonNull(
            Thread.currentThread().getContextClassLoader(), Resources.class.getClassLoader());
...
```

I can change and test if needed.

**Comment by pks-1981** [comment]
> Thanks! We need to discuss the classloader logic itself, but the replacements are good.
> 
> @pks-1981 for the WireMock 3 announcements, do you want to be credited as `@pks-1981` or do you prefer a real name or another alias?
I'm happy with pks-1981)))

**Comment by tomakehurst** [comment]
Thanks!

**Pull request merged**